utils/slices.go

```diff
@@ -1,10 +1,34 @@
 package utils
 
 func Flatten[T any](slices ...[]T) []T {
-	var res []T
+	var result []T
 	for _, slice := range slices {
-		res = append(res, slice...)
+		result = append(result, slice...)
 	}
 
-	return res
+	return result
+}
+
+func Map[T1, T2 any](slice []T1, f func(T1) T2) []T2 {
+	if slice == nil {
+		return nil
+	}
+
+	result := make([]T2, len(slice))
+	for i, v := range slice {
+		result[i] = f(v)
+	}
+
+	return result
+}
+
+func Filter[T any](slice []T, f func(T) bool) []T {
+	var result []T
+	for _, e := range slice {
+		if f(e) {
+			result = append(result, e)
+		}
+	}
+
+	return result
```

Comment:
do we still need these?

Reply:
I left these helpers for future use. They quite helpful